### PR TITLE
Message id support continuation

### PR DESCRIPTION
**Pull request by poljar**
This are fixes that were made as part of #1349. But auto-merge was enabled on that PR before the review phase was done.

### CODE REVIEW

File: crates/matrix-sdk-crypto/src/verification/machine.rs [hunk 1]
```diff
@@ -318,10 +318,6 @@ impl VerificationMachine {
     ) -> Result<(), CryptoStoreError> {
         let event = event.into();
 
-        let span = tracing::Span::current();
-        span.record("event_type", display(event.event_type()));
-        span.record("sender", display(event.sender()));
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/verification/machine.rs:323]: If we no longer need these, they should also not be mentioned in the `instrument` arguments.

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 2]
```diff
@@ -862,24 +861,26 @@ impl OlmMachine {
...
         }
 
-        let message_id = event
-            .deserialize_as::<ToDeviceStub<'_>>()
-            .ok()
-            .and_then(|t| t.content.message_id.into_option());
-
-        tracing::Span::current().record("message_id", message_id);
+        if let Ok(event) = event.deserialize_as::<ToDeviceStub<'_>>() {
+            tracing::Span::current().record("sender", event.sender);
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/machine.rs:877]: IIRC using `field::display` or `%` inside the macros stops tracing from adding quotes around the string or escaping things inside. Do you prefer the quotes? (we are kinda inconsistent about it)

File: crates/matrix-sdk-crypto/src/machine.rs [hunk 3]
```diff
@@ -862,24 +861,26 @@ impl OlmMachine {
...
+            tracing::Span::current().record("event_type", event.event_type);
+            tracing::Span::current().record("message_id", event.content.message_id);
```
Comment (jplatte) [inline crates/matrix-sdk-crypto/src/machine.rs:879]:
```suggestion
            if let Some(message_id) = event.content.message_id {
                tracing::Span::current().record("message_id", message_id);
            }
```

File: crates/matrix-sdk-crypto/src/verification/machine.rs (hunk 1 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/verification/machine.rs:323]: Do we want to keep the span, i.e. the `#[instrument]` attribute?

File: crates/matrix-sdk-crypto/src/machine.rs (hunk 3 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/machine.rs:879]: Hmm, I thought that I need this, but the `None` isn't displayed when I record it unconditionally.

File: crates/matrix-sdk-crypto/src/verification/machine.rs (hunk 1 above)
Comment (jplatte) [reply crates/matrix-sdk-crypto/src/verification/machine.rs:323]: 🤷🏼

File: crates/matrix-sdk-crypto/src/machine.rs (hunk 2 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/machine.rs:877]: Trying this out, I think I prefer them, especially since some things may be empty strings.

Comment (jplatte) [reply crates/matrix-sdk-crypto/src/machine.rs:877]: Right. Maybe we should use debug representation for user IDs and such everywhere then, which IIRC print the quoted string.

**Review by jplatte — APPROVED** [review]

File: crates/matrix-sdk-crypto/src/verification/machine.rs (hunk 1 above)
Comment (poljar) [reply crates/matrix-sdk-crypto/src/verification/machine.rs:323]: Left it in for now. If we decide that it's useless we can remove it later.

**Pull request merged**